### PR TITLE
dimension -> side_dimension where it makes sense

### PR DESCRIPTION
**Pull request by mohamed82008**
Closes https://github.com/JuliaOpt/MathOptInterface.jl/issues/412.

### CODE REVIEW

**Comment by mlubin** [comment]
I'm surprised this didn't require touching any of the tests.

File: src/sets.jl [hunk 1]
```diff
@@ -197,11 +197,11 @@ end
...
 
 """
-    PositiveSemidefiniteConeTriangle(dimension)
+    PositiveSemidefiniteConeTriangle(side_dimension)
 
-The (vectorized) cone of symmetric positive semidefinite matrices, with side length `dimension` and with off-diagonals unscaled.
+The (vectorized) cone of symmetric positive semidefinite matrices, with side length `side_dimension` and with off-diagonals unscaled.
 The entries of the upper triangular part of the matrix are given column by column (or equivalently, the entries of the lower triangular part are given row by row).
-An ``dimension \\times dimension`` matrix has ``dimension(dimension+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has side dimension ``\\sqrt{1/4 + 2 n} - 1/2`` elements.
+An ``side_dimension \\times side_dimension`` matrix has ``side_dimension(side_dimension+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has a ``side_dimension`` of ``\\sqrt{1/4 + 2 n} - 1/2`` rows/columns.
```
[the remark this answers is not in the document]
Comment (mohamed82008) [reply src/sets.jl]: Missed that, thanks, fix coming up.

File: src/sets.jl [hunk 2]
```diff
@@ -197,11 +197,11 @@ end
...
 
 """
-    PositiveSemidefiniteConeTriangle(dimension)
+    PositiveSemidefiniteConeTriangle(side_dimension)
 
-The (vectorized) cone of symmetric positive semidefinite matrices, with side length `dimension` and with off-diagonals unscaled.
+The (vectorized) cone of symmetric positive semidefinite matrices, with side length `side_dimension` and with off-diagonals unscaled.
 The entries of the upper triangular part of the matrix are given column by column (or equivalently, the entries of the lower triangular part are given row by row).
-An ``dimension \\times dimension`` matrix has ``dimension(dimension+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has side dimension ``\\sqrt{1/4 + 2 n} - 1/2`` elements.
+A ``d \\times d`` matrix has ``d(d+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has a side dimension ``d`` of ``\\sqrt{1/4 + 2 n} - 1/2`` rows/columns.
```
Comment (blegat) [inline src/sets.jl]: Need to add "where ``d`` is `side_dimension`"

Comment (mohamed82008) [reply src/sets.jl]: It is pretty clear that it is when you read the whole sentence. I will repeat myself for extra clarity.

File: src/sets.jl [hunk 3]
```diff
@@ -197,11 +197,11 @@ end
...
 
 """
-    PositiveSemidefiniteConeTriangle(dimension)
+    PositiveSemidefiniteConeTriangle(side_dimension)
 
-The (vectorized) cone of symmetric positive semidefinite matrices, with side length `dimension` and with off-diagonals unscaled.
+The (vectorized) cone of symmetric positive semidefinite matrices, with side length `side_dimension` and with off-diagonals unscaled.
 The entries of the upper triangular part of the matrix are given column by column (or equivalently, the entries of the lower triangular part are given row by row).
-An ``dimension \\times dimension`` matrix has ``dimension(dimension+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has side dimension ``\\sqrt{1/4 + 2 n} - 1/2`` elements.
+A ``d \\times d`` matrix has ``d(d+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has a side dimension ``d`` of ``\\sqrt{1/4 + 2 n} - 1/2`` rows/columns, where ``d`` is the input `side_dimension`.
```
Comment (mlubin) [inline src/sets.jl]: Why is `side_dimension` an input in this case? Isn't `n` the input in this sentence? (I.e., it explains how to compute `d` from `n`.)

Comment (mohamed82008) [reply src/sets.jl]: I meant input as in `PositiveSemidefiniteConeTriangle(side_dimension)`. But ya it makes it ambiguous and is unnecessary.

File: src/sets.jl [hunk 4]
```diff
@@ -197,11 +197,11 @@ end
...
 
 """
-    PositiveSemidefiniteConeTriangle(dimension)
+    PositiveSemidefiniteConeTriangle(side_dimension)
 
-The (vectorized) cone of symmetric positive semidefinite matrices, with side length `dimension` and with off-diagonals unscaled.
+The (vectorized) cone of symmetric positive semidefinite matrices, with side length `side_dimension` and with off-diagonals unscaled.
 The entries of the upper triangular part of the matrix are given column by column (or equivalently, the entries of the lower triangular part are given row by row).
-An ``dimension \\times dimension`` matrix has ``dimension(dimension+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has side dimension ``\\sqrt{1/4 + 2 n} - 1/2`` elements.
+A ``d \\times d`` matrix has ``d(d+1)/2`` lower-triangular elements, so for the vectorized cone of dimension ``n``, the corresponding symmetric matrix has a side dimension ``d`` of ``\\sqrt{1/4 + 2 n} - 1/2`` rows/columns, where ``d`` is the `side_dimension`.
```
Comment (mlubin) [inline src/sets.jl]: This sentence is still a bit awkward. Could you restructure it to first say what the point is ("here is how to calculate the side dimension from the flat dimension") and then provide the equations?

Comment (mlubin) [reply src/sets.jl]: I was using this as an opportunity to suggest a small fix to a pre-existing issue with the wording, so you're right that it's orthogonal to this change. Generally it's good to try and fix small things when you notice them up to the point where it becomes annoying. I'll tweak this sentence in your PR.

**Pull request merged**